src/history/types.ts

```diff
@@ -1,18 +1,25 @@
 interface HistoryData {
-  date: string
-  [key: string]: string
+  readonly date: string
+  readonly [key: string]: string
 }
 
 export interface HistoryProps {
   readonly company: string
   readonly history: HistoryData[]
   readonly priceFilter: string
   readonly dateFilter: string
-  onClickFilterHistoryByDate(symbol: string, value: string): void
-  onClickFilterHistoryByPrice(symbol: string, value: string): void
+  readonly onClickFilterHistoryByDate: (symbol: string, value: string) => void
+  readonly onClickFilterHistoryByPrice: (symbol: string, value: string) => void
```

Comment:
onClickFilterHistoryByPrice and onClickFilterHistoryByData have functions of same definition type. I suggest you create a function type ClickFilter: (symbol: string, value:string) => void, and substitute it for functions you have declared.